### PR TITLE
Upgrade muPDF to version 1.22.0

### PR DESCRIPTION
**Pull request by nickolay168**
I have made some minor changes to Orion Viewer to upgrade muPDF.

### CODE REVIEW

File: orion-viewer/version.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-orion.version.code=198
-orion.version.name=0.81.3
\ No newline at end of file
+orion.version.code=199
```
Comment (max-kammerer) [inline orion-viewer/version.properties:1]: I do +4 increment, as there are 4 release builds that are published on Google Play
I will fix it by my own

File: thirdparty_build.gradle [hunk 2]
```diff
@@ -9,8 +9,25 @@ buildscript {
 /*DJVU CHAIN*/
 task downloadDjvu {
     doLast {
-        exec {
-            commandLine = ['git', 'clone', 'git://git.code.sf.net/p/djvu/djvulibre-git', djvu]
+        def djvuFolder = new File(djvu)
+        if (djvuFolder.exists()) {
+            // Clean previous checkout
+            exec {
+                commandLine = ['git', '-C', djvu, 'checkout', '.']
```
Comment (max-kammerer) [inline thirdparty_build.gradle:16]: This is quite unusual approach for me to clean previous checkout. (I wonder what is origin of this approach?)
Why don't use just clean or reset?

Looks like that first two steps could be combined, couldn't it?

File: thirdparty_build.gradle [hunk 3]
```diff
@@ -9,8 +9,25 @@ buildscript {
 /*DJVU CHAIN*/
 task downloadDjvu {
     doLast {
```
Comment (max-kammerer) [inline thirdparty_build.gradle:11]: Why do you put new exec instuction in `doLast` and kept 
```
        exec {
            workingDir file(djvu)
            commandLine = ['git', 'checkout', '32ff0948c5bc27c0b7507dba90e726fc941fcc42'/*current master*/]
        }
 ```
 out of it?

File: thirdparty_build.gradle [hunk 4]
```diff
@@ -9,8 +9,25 @@ buildscript {
...
+            exec {
+                commandLine = ['git', '-C', djvu, 'checkout', '.']
+            }
+            // Switch to master
+            exec {
+                commandLine = ['git', '-C', djvu, 'checkout', 'master']
+            }
+            // Pull
+            exec {
+                commandLine = ['git', '-C', djvu, 'pull']
```
Comment (max-kammerer) [inline thirdparty_build.gradle:24]: Probably simple `fetch` would be enough instead of `pull`

File: nativeLibs/mupdfModule/jni/CMakeLists.txt [hunk 5]
```diff
@@ -254,7 +319,7 @@ ${MUPDF_PATH}/thirdparty/zlib
 #-Dhb_realloc_impl=fz_hb_realloc \
 #-Dhb_free_impl=fz_hb_free \
 
-target_compile_definitions(mupdf_thirdparty PRIVATE -DFT2_BUILD_LIBRARY -DDARWIN_NO_CARBON -DFT_CONFIG_MODULES_H="slimftmodules.h" -DFT_CONFIG_OPTIONS_H="slimftoptions.h" -DHAVE_STDINT_H -DOPJ_STATIC -DOPJ_HAVE_INTTYPES_H -DOPJ_HAVE_STDINT_H -DHAVE_FALLBACK=1 -DHAVE_OT -DHAVE_UCDN -DHB_NO_MT -Dhb_malloc_impl=fz_hb_malloc -Dhb_calloc_impl=fz_hb_calloc -Dhb_realloc_impl=fz_hb_realloc -Dhb_free_impl=fz_hb_free)
+target_compile_definitions(mupdf_thirdparty PRIVATE -DFT2_BUILD_LIBRARY -DDARWIN_NO_CARBON -DFT_CONFIG_MODULES_H="slimftmodules.h" -DFT_CONFIG_OPTIONS_H="slimftoptions.h" -DHAVE_STDINT_H -DOPJ_STATIC -DOPJ_HAVE_INTTYPES_H -DOPJ_HAVE_STDINT_H -DHAVE_FALLBACK=1 -DHAVE_OT -DHAVE_UCDN -DHAVE_FREETYPE -DHB_NO_MT -Dhb_malloc_impl=fz_hb_malloc -Dhb_calloc_impl=fz_hb_calloc -Dhb_realloc_impl=fz_hb_realloc -Dhb_free_impl=fz_hb_free)
```
Comment (max-kammerer) [inline nativeLibs/mupdfModule/jni/CMakeLists.txt:322]: Why did you add -DHAVE_FREETYPE flag?

**Comment by max-kammerer** [comment]
@nickolay168 I've merged part of this request in https://github.com/max-kammerer/orion-viewer/commit/8a4a01191c4c17b374c8abfccb831130c4713bd4

There are still some question above for remaining part

File: nativeLibs/mupdfModule/jni/CMakeLists.txt (hunk 5 above)
Comment (nickolay168) [reply nativeLibs/mupdfModule/jni/CMakeLists.txt:322]: If we will remove this flag, we will get an error about missing function. That is happening because in new version of harfbuzz dependency we now have the ifdef condition: https://github.com/ArtifexSoftware/thirdparty-harfbuzz/blob/07c4f02f8c39f4227887249afbd193859da34a68/src/hb-ft.cc#L32C1-L32C21, so if we will remove this flag, the linking step will fail.

File: thirdparty_build.gradle (hunk 2 above)
Comment (nickolay168) [reply thirdparty_build.gradle]: We are working in the detached head here and just pulling will result in the error, I used checkout to move to HEAD. I have fixed this hack.

File: thirdparty_build.gradle (hunk 4 above)
Comment (nickolay168) [reply thirdparty_build.gradle]: Fixed

File: thirdparty_build.gradle (hunk 3 above)
Comment (nickolay168) [reply thirdparty_build.gradle:11]: Here I am trying to handle the edge case if we are trying to checkout the commit later then the one known currently checked out repository. So we first move HEAD by fetching the latest master and then switching to the pinned commit. So we either clone or fetch master before switching to detached HEAD.

File: orion-viewer/version.properties (hunk 1 above)
Comment (nickolay168) [reply orion-viewer/version.properties]: Fixed

File: thirdparty_build.gradle [hunk 6]
```diff
@@ -11,17 +11,14 @@ task downloadDjvu {
...
+            // detached.
             exec {
-                commandLine = ['git', '-C', djvu, 'checkout', '.']
-            }
-            // Switch to master
-            exec {
-                commandLine = ['git', '-C', djvu, 'checkout', 'master']
+                commandLine = ['git', '-C', djvu, 'checkout', '-f', 'master']
             }
             // Pull
```
Comment (max-kammerer) [inline thirdparty_build.gradle:19]: incorrect comment

Comment (nickolay168) [reply thirdparty_build.gradle]: Fixed.

**Comment by max-kammerer** [comment]
@nickolay168, @nick863  Thank you! Merged manually with small changes
